src/main/default/classes/TriggerHandler.cls

```diff
@@ -65,51 +73,66 @@ public class TriggerHandler {
-        disabledObjects.clear();
-        disabledHandlers.clear();
-        disabledObjectHandlers.clear();
-        for (Trigger_Control__mdt triggerControl : triggerControls) {
-            String objectName = triggerControl.Object__c == null ? null : triggerControl.Object__c.toLowerCase();
-            String handlerName = triggerControl.Class__c == null ? null : triggerControl.Class__c.toLowerCase();
-            if (handlerName == null && objectName != null) {
-                disabledObjects.add(objectName);
-            } else if (handlerName != null && objectName == null) {
-                disabledHandlers.add(handlerName);
-            } else if (handlerName != null && objectName != null) {
-                if (!disabledObjectHandlers.containsKey(objectName)) {
-                    disabledObjectHandlers.put(objectName, new Set<String>());
-                }
-                disabledObjectHandlers.get(objectName).add(handlerName);
+    private void doAutomaticBinding() {
+        if (!triggerConfig.enabled) {
+            return;
+        }
+        for (String handlerName : triggerConfig.handlerOrder) {
+            TriggerConfigService.HandlerConfig handlerConfig = triggerConfig.handlers.get(handlerName);
+            // do not bind if auto bind is off or if handler is disabled
+            if (!handlerConfig.autoBind || !handlerConfig.enabled) {
+                continue;
+            }
+            Type t = getType(handlerConfig.className);
+            if (t == null) {
+                System.debug(handlerConfig.className + '? Looks like that handler class does not exist!');
+                continue;
+            }
+            HandlerInterface handlerClass = (HandlerInterface) t.newInstance();
+            if (handlerConfig.beforeInsert) {
+                bind(Evt.beforeInsert, handlerClass);
+            }
+            if (handlerConfig.afterInsert) {
+                bind(Evt.afterInsert, handlerClass);
+            }
+            if (handlerConfig.beforeUpdate) {
+                bind(Evt.beforeUpdate, handlerClass);
+            }
+            if (handlerConfig.afterUpdate) {
+                bind(Evt.afterUpdate, handlerClass);
+            }
+            if (handlerConfig.beforeDelete) {
+                bind(Evt.beforeDelete, handlerClass);
+            }
+            if (handlerConfig.afterDelete) {
+                bind(Evt.afterDelete, handlerClass);
+            }
+            if (handlerConfig.afterUndelete) {
+                bind(Evt.afterUndelete, handlerClass);
             }
         }
     }
-    static {
-        cacheTriggerControl([SELECT Class__c, Object__c FROM Trigger_Control__mdt]);
-    }
 
-    private Boolean objectDisabled(String xSObjType) {
-        return disabledObjects.contains(xSObjType);
+    private Type getType(String className) {
+        Type t;
+        try {
+            t = Type.forName(className); // doesn't throw exception if invalid name!
```

Comment:
if this doesn't throw an exception why do we have the try/catch?

Reply:
Got rid of it. Was trying to think if there are any other errors or anything we should add to this method. Expected it to throw an error which is why I put it there in the first place, but it surprisingly just returns null if you pass it something invalid? 🤷